skbio/alignment/_path.py

```diff
@@ -0,0 +1,308 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio._base import SkbioObject
+from skbio.util._decorator import overrides, classonlymethod
+
+
+class AlignPath(SkbioObject):
```

Comment:
Is it worthwhile to flesh out a practical use-case of this object? Pseudocode in the issue tracker may be useful to help us understand how this code is going to be used (i.e. pseudocount for progressive alignment)

Reply:
@qiyunzhu You may have a clearer or more well defined use case in mind than I do?

Reply:
I think we should do it but it is outside the scope of the current PR. Specifically, we can:
1. Generate a pairwise alignment using external Python libraries like BioPython or Biotite and convert the result into `AlignPath`.
2. Generate a multiple alignment using external programs like ClustalO or MAFFT and read the result (multi-fasta format) into `AlignPath`.
3. Perform pairwise or multiple alignment using scikit-bio iteself (to be re-modeled) and return an `AlignPath` object.
4. Manually create a plausible alignment and calculate its alignment score.

There may be variants of each case. For example, for 3, we can return all optimal alignments instead of just one arbitrary alignment (pseudocode):

```python
for aln in pairwise_align(seq1, seq2, return_all=True):
    scores.append(aln.align_score())
```